src/serena/agent.py

```diff
@@ -536,6 +536,49 @@ def apply(
         return self._limit_length(result, max_answer_chars)
 
 
+class GetReferencingCodeExtractsTool(Tool):
+    """
+    Gets the code blocks that reference the symbol at the given location.
+    """
+
+    def apply(
+        self,
+        relative_path: str,
+        line: int,
+        column: int,
+        context_lines_before: int = 0,
+        context_lines_after: int = 0,
+        max_answer_chars: int = _DEFAULT_MAX_ANSWER_LENGTH,
+    ) -> str:
+        """
+        Returns short code extracts where the symbol at the given location is referenced.
+
+        Contrary to the `find_referencing_symbols` tool, this tool returns references that are not symbols but instead
+        code extracts that may or may not be contained in a symbol (for example, file-level calls).
+        It may make sense to use this tool if you want to get a quick and dirty overview of the code that references
```

Comment:
I don't really think there is anything "dirty" about this. Depending on the use case, you might not care at all about the full context (symbol retrieval could only give you the symbol's full body) but only about the very concrete usage (e.g. is the function being called with some parameter or not). I think it's actually a very reasonable way of looking at usages (which I constantly make use of in IDEs).
So I'd lose the "and dirty" part.

Reply:
Done